tests/components/device_tracker/test_ee_brightbox.py

```diff
@@ -0,0 +1,102 @@
+"""Tests for the EE BrightBox device scanner."""
+from datetime import datetime, timedelta
+from asynctest import patch
+
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, ee_brightbox)
+from homeassistant.const import (
+    CONF_PLATFORM, CONF_USERNAME, CONF_PASSWORD, CONF_HOST)
+
+
+def _get_devices_return_value():
+    return [
+        {
+            'mac': 'AA:BB:CC:DD:EE:FF',
+            'ip': '192.168.1.10',
+            'hostname': 'hostnameAA',
+            'activity_ip': True,
+            'port': 'eth0',
+            'time_last_active': datetime.now() - timedelta(minutes=5),
+        },
+        {
+            'mac': '11:22:33:44:55:66',
+            'hostname': 'hostname11',
+            'ip': '192.168.1.11',
+            'activity_ip': True,
+            'port': 'wl0',
+            'time_last_active': datetime.now() - timedelta(hours=5),
+        },
+        {
+            'mac': 'FF:FF:FF:FF:FF:FF',
+            'hostname': 'hostnameFF',
+            'ip': '192.168.1.12',
+            'activity_ip': False,
+            'port': 'wl1',
+            'time_last_active': datetime.now() - timedelta(days=5),
+        }
+    ]
+
+
+def _configure_scanner(eebrightbox_mock):
+    config = {
+        CONF_PASSWORD: 'password_test',
+        'version': 2,
+    }
+
+    eebrightbox_instance = eebrightbox_mock.return_value
+    eebrightbox_instance.__enter__.return_value = eebrightbox_instance
+    eebrightbox_instance.get_devices.return_value = _get_devices_return_value()
+
+    return ee_brightbox.EEBrightBoxScanner(config)
+
+
+async def test_get_scanner_returns_instance(hass):
+    """Test get scanner."""
+    config = {
+        ee_brightbox.DOMAIN: PLATFORM_SCHEMA({
+            CONF_PLATFORM: ee_brightbox.DOMAIN,
+            CONF_HOST: '192.168.1.1',
+            CONF_USERNAME: 'username_test',
+            CONF_PASSWORD: 'password_test',
+            'version': 2,
+        })
+    }
+
+    scanner = ee_brightbox.get_scanner(hass, config)
```

Comment:
Don't call `get_scanner` directly. Set up the device tracker component and the correct platform, while patching appropriately to avoid I/O. Let the platform and component create the entity and then assert the state of the entity by getting that from the core state machine.